### PR TITLE
etcd-cpp-apiv3 0.2.3 (new formula)

### PR DESCRIPTION
**Pull request by sighingnow**
Signed-off-by: Tao He <sighingnow@gmail.com>

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by sighingnow** [comment]
Hi @chenrui333 @carlocab I have fixed previous CI failure and upgrade a new version for this formula. Could you folks please help me to approve and rerun the check so that we can get this PR landed?

Thanks in advance!

**Comment by sighingnow** [comment]
Could we get this formula merged?

Thanks!

File: Formula/etcd-cpp-apiv3.rb [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    # to `brew install` such as `--HEAD` also need to be provided to `brew test`.
+    #
+    # The installed folder is not in the path, so use the entire path to any
+    # executables being tested: `system "#{bin}/program", "do", "something"`.
+    (testpath/"test.cc").write <<~EOS
+      #include <iostream>
+      #include <etcd/Client.hpp>
+
+      int main() {
+        etcd::Client etcd("http://127.0.0.1:2379");
```
Comment (SMillerDev) [inline Formula/etcd-cpp-apiv3.rb:51]: can we use `free_port` here to pick a random free port?

File: Formula/etcd-cpp-apiv3.rb [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+  test do
+    # `test do` will create, run in and delete a temporary directory.
+    #
+    # This test will fail and we won't accept that! For Homebrew/homebrew-core
+    # this will need to be a test that verifies the functionality of the
+    # software. Run the test with `brew test etcd-cpp-apiv3`. Options passed
+    # to `brew install` such as `--HEAD` also need to be provided to `brew test`.
+    #
+    # The installed folder is not in the path, so use the entire path to any
+    # executables being tested: `system "#{bin}/program", "do", "something"`.
```
Comment (SMillerDev) [inline Formula/etcd-cpp-apiv3.rb:45]:
```suggestion
```

File: Formula/etcd-cpp-apiv3.rb (hunk 1 above)
Comment (sighingnow) [reply Formula/etcd-cpp-apiv3.rb:42]: Fixed.

Comment (sighingnow) [reply Formula/etcd-cpp-apiv3.rb]: > can we use `free_port` here to pick a random free port?

Fixed.

**Comment by sighingnow** [comment]
Syntax lint error fixed.

**Review by SMillerDev — APPROVED** [review]
Thanks @sighingnow  ! Without contributions like yours it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉

**Comment by sighingnow** [comment]
> Thanks @sighingnow ! Without contributions like yours it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉

Thank you @SMillerDev !

**Pull request closed without merging**